### PR TITLE
Screen 6 deal repayment

### PR DESCRIPTION
**Pull request by BartDeCaluwe**
This PR will add:

- RepayDeal form with validation
- Deal repayment screen that enables the borrower to repay both interest and principal
- Repay button on the deals table that redirects to the repay screen
- Deal utils that calculate various ratios + tests for these utils
- DealCard component that contains the shared markup of the `/show` and `/repay` page. + Storybook story
- SideMenu layout to the deal show page
- Deal table row is not fully clickable anymore, to accommodate the repay button.

## Checklist

- [x] I used the [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/) specification.
- [x] Unit tests have been created if a new feature was added.

### CODE REVIEW

File: src/utils/deal.utils.ts [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+export const calculateMonthlyRepaymentAmount = (deal: Deal) => {
+	if (!deal) {
+		return;
+	}
+
+	return toUIAmount(new Big(deal.totalInterest.toNumber() / (deal.timeToMaturity / 30))).toNumber();
+};
+
+export const calculateInterestRepaidRatio = (deal: Deal) => {
```
Comment (RuvendeGroote) [inline src/utils/deal.utils.ts:13]: client candidates?

File: src/utils/deal.utils.ts [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+	return new Ratio(
+		toUIAmount(new Big(deal.interestRepaid)).toNumber(),
+		toUIAmount(deal.totalInterest).toNumber()
+	);
+};
+
+export const calculatePrincipalRepaidRatio = (deal: Deal) => {
+	return new Ratio(
+		toUIAmount(new Big(deal.principalAmountRepaid)).toNumber(),
```
Comment (RuvendeGroote) [inline src/utils/deal.utils.ts]: why the conversion to uiamount here? it doesn't change the ratio or anything

File: src/utils/deal.utils.ts [hunk 3]
```diff
@@ -0,0 +1,29 @@
+import { Deal, Ratio } from "@credix/credix-client";
+import { toUIAmount } from "@utils/format.utils";
+import Big from "big.js";
+
+export const calculateMonthlyRepaymentAmount = (deal: Deal) => {
+	if (!deal) {
+		return;
+	}
+
+	return toUIAmount(new Big(deal.totalInterest.toNumber() / (deal.timeToMaturity / 30))).toNumber();
```
Comment (RuvendeGroote) [inline src/utils/deal.utils.ts]: use big for all divisions

File: src/pages/[marketplace]/deals/index.tsx [hunk 4]
```diff
@@ -63,6 +30,73 @@ const Deals: NextPageWithLayout = () => {
...
+	const { publicKey } = useWallet();
+
+	const dealsTableColumns: ColumnsProps[] = [
+		{
+			title: "Name",
+			icon: "stacked-column-down",
+			dataIndex: "name",
+			key: "name",
+			ellipsis: true,
+			className: "hover:cursor-pointer hover:bg-neutral-10",
```
Comment (RuvendeGroote) [inline src/pages/[marketplace]/deals/index.tsx]: this column header looks clickable but it isn't

**Review by RuvendeGroote — CHANGES_REQUESTED** [review]

File: src/utils/deal.utils.ts (hunk 1 above)
Comment (BartDeCaluwe) [reply src/utils/deal.utils.ts:13]: Yes!

Comment (RuvendeGroote) [reply src/utils/deal.utils.ts:13]: made a ticket

File: src/utils/deal.utils.ts [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+export const calculateMonthlyRepaymentAmount = (deal: Deal) => {
+	if (!deal) {
+		return;
+	}
+
+	return toUIAmount(deal.totalInterest.div(new Big(deal.timeToMaturity).div(30))).toNumber();
+};
+
+export const calculateInterestRepaidRatio = (deal: Deal) => {
+	return new Ratio(new Big(deal.interestRepaid).toNumber(), deal.totalInterest.toNumber());
```
Comment (RuvendeGroote) [inline src/utils/deal.utils.ts]: interestRepaid is already a number

Comment (RuvendeGroote) [reply src/utils/deal.utils.ts]: same below

**Review by RuvendeGroote — CHANGES_REQUESTED** [review]

**Review by RuvendeGroote — APPROVED** [review]

**Review by BouweCeunen — APPROVED** [review]

**Pull request merged**